index.d.ts

```diff
@@ -0,0 +1,89 @@
+declare module "moleculer-apollo-server" {
+	import { ServiceSchema, Context } from "moleculer";
+	import { Config } from "apollo-server-core";
+	import { OptionsUrlencoded } from "body-parser";
+	import { SchemaDirectiveVisitor, IResolvers } from "graphql-tools";
+
+	export {
+		GraphQLUpload,
+		GraphQLExtension,
+		gql,
+		ApolloError,
+		toApolloError,
+		SyntaxError,
+		ValidationError,
+		AuthenticationError,
+		ForbiddenError,
+		UserInputError,
+		defaultPlaygroundOptions,
+	} from "apollo-server-core";
+
+	export * from "graphql-tools";
+
+	export interface ApolloServerOptions {
+		path: string;
+		disableHealthCheck: boolean;
+		onHealthCheck: () => {};
+	}
+
+	export class ApolloServer {
+		createGraphQLServerOptions(req: any, res: any): Promise<any>;
+		createHandler(options: ApolloServerOptions): void;
+		supportsUploads(): boolean;
+		supportsSubscriptions(): boolean;
+	}
+
+	export interface ActionResolverSchema {
+		action: string;
+		rootParams?: {
+			[key: string]: string;
+		};
+		dataLoader?: boolean;
+	}
+
+	export interface ServiceResolverSchema {
+		[key: string]: {
+			[key: string]: ActionResolverSchema;
+		};
+	}
+
+	export interface ServiceRouteOptions {
+		path: string;
+		use?: any[];
+		etag?: boolean;
+		whitelist?: string[];
+		authorization?: boolean;
+		camelCaseNames?: boolean;
+		aliases?: {
+			[key: string]: any; // Should discuss more on this. string | AliasSchema, ...
+		};
+		bodyParsers?: {
+			json: boolean;
+			urlencoded: OptionsUrlencoded;
+		};
+		cors?: boolean | { origin: string[]; methods: string[] };
```

Comment:
Running in my own repo, I hit a type issue because `exposedHeaders` is not defined here.  In looking at moleculer-web docs, I see `allowedHeaders`, `exposedHeaders`, `credentials`, and `maxAge` as properties that could be set for the `cors` property.

Reply:
I also believe that each of these properties is optional, so the properties should be suffixed with `?`.

Reply:
I've checked in `moleculer.sevices` docs. It was presented. Updated with separated interface `ServiceRouteCorsOptions`